lib/spack/spack/cmd/testsuite.py

```diff
@@ -0,0 +1,221 @@
+            compilers = yamlFile['compilers']
+            
+            try:
+                path=yamlFile['path']
+            except KeyError as e:
+                tty.msg("Path not found, will use " + str(spack.prefix)+"/var/spack/cdash/")
+
+            try:
+                dashboards = yamlFile['dashboard']
+            except KeyError as e:
+                if path is "":
+                    tty.msg("dashboard url was not found. xml files will be stored at " + str(spack.prefix)+"/var/spack/cdash/")
+                else:
+                    tty.msg("dashboard url was not found. xml files will be stored at path provided.")
+    #creating a list of packages 
+    for package in packages:
+        for pkg in package:
+            if pkg in enabledTests:
+                versions = package[pkg][0]['versions']
+                for version in versions:
+                    # producing packages at available versions. Sample file contains only checksum'd
+                    packageVersion.append(str(pkg)+"@"+str(version))
+
+    #creating a list of compilers
+    for compiler in compilers:
+        for comp in compiler:
+            versions = compiler[comp][0]['versions']
+            for version in versions:
+                # producing compilers at available versions. Sample file contains only checksum'd
+                compilerVersion.append(str(comp)+"@"+str(version))
+    #reducing compiler list to whats actually available on the system
+    compilerVersion = reduceCompilerList(compilerVersion)
+    #producing a list of tests with a combination of packages and compilers
+    for pkg in packageVersion:
+        for comp in compilerVersion:
+            tests.append(str(pkg)+"%"+str(comp))
+
+    #loading test excusions
+    exclusions = yamlFile['exclusions']
+    if len(exclusions) != 0:
+        #remove test that match the exclusion
+        for exclusion in exclusions:
+            tests=removeTests(tests, exclusion)
+
+
+    concreteTests = []
+    
+    #setting up tests for contretizing
+    for test in tests:
+        spec = Spec(test)
+        #uninstall all packages before installing. This will reduce the number of skipped package installs.
+        while (len(spack.store.db.query(spec)) > 0):
+            spec.concretize()
+            tty.msg("uninstalling " + str(spec))
+            pkg = spack.repo.get(spec)
+            pkg.do_uninstall()
+        #concretize, failing can occur if the package uses the wrong compiler which would produce a failure for cdash
+        try:
+            spec.concretize()
+            concreteTests.append(spec.to_yaml())
+            parser = argparse.ArgumentParser()
+            install.setup_parser(parser)
+            args = parser.parse_args([cdash]) #use cdash-complete if you want configure, build and test output.
+            args.package = test
+            install.install(parser, args)
+        except Exception as ex:
```

Comment:
Whatever error occurs should probably go in the test log and get displayed as part of CDash output.